src/m365/planner/commands/task/task-set.ts

```diff
@@ -0,0 +1,510 @@
+          return Promise.reject(`Cannot proceed with planner task update. The following users provided are invalid : ${invalidUsers.join(',')}`);
+        }
+        return Promise.resolve(userIds);
+      });
+  }
+
+  private getBucketId(options: Options): Promise<string | undefined> {
+    if (options.bucketId) {
+      return Promise.resolve(options.bucketId);
+    }
+
+    if (!options.bucketName) {
+      return Promise.resolve(undefined);
+    }
+
+    return this
+      .getPlanId(options)
+      .then(planId => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: PlannerBucket[] }>(requestOptions);
+      })
+      .then((response) => {
+        const bucket: PlannerBucket | undefined = response.value.find(val => val.name === options.bucketName);
+
+        if (!bucket) {
+          return Promise.reject(`The specified bucket does not exist`);
+        }
+
+        return Promise.resolve(bucket.id as string);
+      });
+  }
+
+  private getPlanId(options: Options): Promise<string> {
+    if (options.planId) {
+      return Promise.resolve(options.planId);
+    }
+
+    return this
+      .getGroupId(options)
+      .then((groupId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans?$filter=(owner eq '${groupId}')`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: PlannerPlan[] }>(requestOptions);
+      }).then((response) => {
+        const plan: PlannerPlan | undefined = response.value.find(val => val.title === options.planName);
+
+        if (!plan) {
+          return Promise.reject(`The specified plan does not exist`);
+        }
+
+        return Promise.resolve(plan.id as string);
+      });
+  }
+
+  private getGroupId(options: Options): Promise<string> {
+    if (options.ownerGroupId) {
+      return Promise.resolve(options.ownerGroupId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: Group[] }>(requestOptions)
+      .then(response => {
+        const group: Group | undefined = response.value[0];
+
+        if (!group) {
+          return Promise.reject(`The specified owner group does not exist`);
+        }
+
+        return Promise.resolve(group.id as string);
+      });
+  }
+
+  private mapRequestBody(options: Options, appliedcategories: { [category: string]: boolean }): any {
```

Comment:
When I inspect the Graph Type for PlannerAppliedCategories then you can see that is an empty interface

```
// tslint:disable-next-line: no-empty-interface 
export interface PlannerAppliedCategories {}
```

So I don't think I can use this one.

Reply:
Thank you for checking. In that case, let's define it as a separate type that we can reference from multiple locations so that we can ensure its consistency across all references.